numpy/f2py/crackfortran.py

```diff
@@ -3231,8 +3231,13 @@ def vars2fortran(block, vars, args, tab='', as_interface=False):
             show(vars)
             outmess('vars2fortran: No definition for argument "%s".\n' % a)
             continue
-        if a == block['name'] and not block['block'] == 'function':
-            continue
+        if a == block['name']:
+            if block['block'] == 'function':
```

Comment:
Maybe something like
```
            if (block['block'] != 'function'
                    or block['block'] == 'function' and block.get('result')):
                continue
```
would make the logic a bit clearer.

Reply:
Fixed.